central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -54,23 +90,165 @@ func (m *managerImpl) ProcessComplianceOperatorInfo(ctx context.Context, complia
+	}
+	if found {
+		return nil, errors.Errorf("Scan configuration named %q already exists.", scanRequest.GetScanName())
+	}
+
+	scanRequest.Id = uuid.NewV4().String()
+	scanRequest.CreatedTime = types.TimestampNow()
+	err = m.scanSettingDS.UpsertScanConfiguration(ctx, scanRequest)
+	if err != nil {
+		log.Error(err)
+		return nil, errors.Errorf("Unable to save scan configuration named %q.", scanRequest.GetScanName())
+	}
+
+	var profiles []string
+	for _, profile := range scanRequest.GetProfiles() {
+		profiles = append(profiles, profile.GetProfileName())
+	}
+
+	for _, clusterID := range clusters {
+		// id for the request message to sensor
+		sensorRequestID := uuid.NewV4().String()
+
+		sensorMessage := &central.MsgToSensor{
+			Msg: &central.MsgToSensor_ComplianceRequest{
+				ComplianceRequest: &central.ComplianceRequest{
+					Request: &central.ComplianceRequest_ApplyScanConfig{
+						ApplyScanConfig: &central.ApplyComplianceScanConfigRequest{
+							Id: sensorRequestID,
+							ScanRequest: &central.ApplyComplianceScanConfigRequest_ScheduledScan_{
+								ScheduledScan: &central.ApplyComplianceScanConfigRequest_ScheduledScan{
+									ScanSettings: &central.ApplyComplianceScanConfigRequest_BaseScanSettings{
+										ScanName:       scanRequest.GetScanName(),
+										StrictNodeScan: true,
+										Profiles:       profiles,
+									},
+									Cron: cron,
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		err := m.sensorConnMgr.SendMessage(clusterID, sensorMessage)
+		var status string
+		if err != nil {
+			status = err.Error()
+			log.Errorf("error sending compliance scan config to cluster %q: %v", clusterID, err)
+		} else {
+			// Request was not rejected so add it to map awaiting response
+			m.trackSensorRequest(sensorRequestID, clusterID, scanRequest.GetId())
+		}
+
+		// Update status in DB
+		err = m.scanSettingDS.UpdateClusterStatus(ctx, scanRequest.GetId(), clusterID, status)
+		if err != nil {
+			log.Error(err)
+			return nil, errors.Errorf("Unable to save scan configuration status for scan named %q.", scanRequest.GetScanName())
+		}
+	}
+
+	return scanRequest, nil
+}
+
+// trackSensorRequest adds sensor request to a map with cluster and scan config that was sent for correlating responses.
+func (m *managerImpl) trackSensorRequest(sensorRequestID, clusterID, scanConfigID string) {
+	m.runningRequestsLock.Lock()
+	defer m.runningRequestsLock.Unlock()
+
+	// Request was not rejected so add it to map awaiting response
+	m.runningRequests[sensorRequestID] = clusterScan{
+		clusterID: clusterID,
+		scanID:    scanConfigID,
+	}
+}
+
+// HandleScanRequestResponse processes response of compliance scan configuration from a sensor.
+func (m *managerImpl) HandleScanRequestResponse(ctx context.Context, requestID string, clusterID string, responsePayload string) error {
+	if !features.ComplianceEnhancements.Enabled() {
+		return nil
```

Comment:
We swallow this because it's ultimately coming from sensor and not an end-user directly?

Reply:
yeah, that may be useful.